perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3032,6 +3949,52 @@ public void removeGroupUnion(PerunSession sess, Group resultGroup, Group operand
 		}
 	}
 
+	@Override
+	public boolean isGroupInStructureSynchronizationTree(PerunSession session, Group group) throws InternalErrorException {
```

Comment:
What if group itself is root of synchronized tree? I would recommend to create two methods - 1] which will search through all nodes except root node and return true or false, 2] which will search in the root node and then call first method. And then I would use these two methods in code where is specific implementation needed.